source/AI.cpp

```diff
@@ -3356,35 +3356,122 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 	}
 	else if(activeCommands.Has(Command::BOARD))
 	{
-		// Board the nearest disabled ship, focusing on hostiles before allies. Holding
-		// `Shift` results in boarding only player-owned escorts in need of assistance.
+		// Depending on the preference, board the (nearest/most expensive/mix of both) disabled ship.
+		// Pressing the board key multiple times cycles between eligible targets.
+		// It will be focusing on hostiles before allies. Holding `Shift`
+		// results in boarding only player-owned escorts in need of assistance.
 		shared_ptr<const Ship> target = ship.GetTargetShip();
-		if(!target || !CanBoard(ship, *target) || (shift && !target->IsYours()))
+		if(!target || (target && activeCommands.Has(Command::WAIT)) || !CanBoard(ship, *target)
+				|| (shift && !target->IsYours()))
 		{
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
 
-			double closest = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
-			bool foundAnything = false;
-			for(const shared_ptr<Ship> &other : ships)
-				if(CanBoard(ship, *other))
+			const auto targetPriority = Preferences::BoardingSetting();
+			enum class BOARDING_PRIORITY{DISTANCE, COST, MIXED};
+			const BOARDING_PRIORITY boardingPriority = targetPriority == "proximity" ? BOARDING_PRIORITY::DISTANCE :
+				targetPriority == "value" ? BOARDING_PRIORITY::COST : BOARDING_PRIORITY::MIXED;
+
+			auto strategy = [&]() noexcept -> function<double(Ship &)>
+			{
+				Point current = ship.Position();
+				switch(boardingPriority)
 				{
-					if(shift && !other->IsYours())
-						continue;
+					case BOARDING_PRIORITY::COST:
+						return [this, &ship](Ship &other) noexcept -> double
+						{
+							// Use the exact cost if the ship was scanned, otherwise use an estimation.
+							return this->Has(ship, other.shared_from_this(), ShipEvent::SCAN_OUTFITS) ?
+								other.Cost() : (other.ChassisCost() * 2.);
+						};
+					case BOARDING_PRIORITY::MIXED:
+						return [this, &ship, current](Ship &other) noexcept -> double
+						{
+							double cost = this->Has(ship, other.shared_from_this(), ShipEvent::SCAN_OUTFITS) ?
+								other.Cost() : (other.ChassisCost() * 2.);
+							return cost * cost / current.DistanceSquared(other.Position());
```

Comment:
While very unlikely, is it not possible for the distance squared to be 0? Maybe just add 0.1 to it?
```suggestion
							return cost * cost / (current.DistanceSquared(other.Position()) + 0.1);
```

Reply:
Funny caus everyone is worried about that xD
I may have to add a comment to say that:

Dividing by zero just means infinity in a double, which is actually understood as such, and does not cause crashes or any issues. Please, try it out!

Reply:
My Googling suggested that the result of dividing by zero in C++ is actually undefined because the specifics are hardware dependent.

Reply:
https://en.cppreference.com/w/cpp/language/operator_arithmetic

This page seems quite explicit that it gives +/-inf, so I guess it's fine?

Reply:
https://en.cppreference.com/w/cpp/language/operator_arithmetic

This page seems quite explicit that it gives +/-inf, so I guess it's fine?